src/cmd/moby/run_qemu.go

```diff
@@ -73,18 +91,38 @@ func runQemu(args []string) {
 	} else {
 		qemuArgs = append(qemuArgs, "-enable-kvm")
 		qemuArgs = append(qemuArgs, "-machine", "q35,accel=kvm:tcg")
+		if container {
+			dockerArgs = append(dockerArgs, "", "")
+			copy(dockerArgs[5:], dockerArgs[3:])
+			dockerArgs[3] = "--device"
+			dockerArgs[4] = "/dev/kvm"
+		}
+	}
+
+	var qemuCmd *exec.Cmd
+	var qemuImgCmd *exec.Cmd
+	if container {
+		dockerPath, err := exec.LookPath("docker")
+		if err != nil {
+			log.Fatal("Unable to find docker in the $PATH")
+		}
+		binArgs := append(dockerArgs, qemuBinPath)
+		imgArgs := append(dockerArgs, qemuImgPath)
+		qemuCmd = exec.Command(dockerPath, binArgs...)
+		qemuImgCmd = exec.Command(dockerPath, imgArgs...)
+	} else {
+		qemuCmd = exec.Command(fullQemuPath)
+		qemuImgCmd = exec.Command(fullQemuImgPath)
```

Comment:
this also looks odd, you have a conditional above for container, and then another just after. How about just one conditional where you set up the args either for local or container and then execute it without conditional?